client/client.go

```diff
@@ -227,6 +164,15 @@ func (c *Client) do(ctx context.Context, method, path string, addr stdaddr.Addre
 		return fmt.Errorf("unmarshal response body: %w", err)
 	}
 
+	type vspResponse struct {
+		Request []byte `json:"request"`
+	}
+
+	// Verify initial request matches server.
+	if !bytes.Equal(reqBody, resp.(vspResponse).Request) {
+		return fmt.Errorf("server response contains differing request")
+	}
```

Comment:
currently in wallet, all of the POST methods do this check, but we do not perform it for GET /api/v3/vspinfo.  So this looks fine as long as that is tested and works.

Reply:
Good catch. I had thoroughly tested all of the POST funcs using v3tool, but not the GET one. I've pulled that logic out of do() up into post()

Reply:
Opened #364 to implement this.